C/c4DocEnumerator.cc

```diff
@@ -36,89 +35,51 @@ CBL_CORE_API const C4EnumeratorOptions kC4DefaultEnumeratorOptions = {
 };
 
 
-typedef function<bool(const Record&, uint32_t/*C4DocumentFlags*/ documentFlags)> EnumFilter;
-
-
-struct C4DocEnumerator: fleece::InstanceCounted {
+struct C4DocEnumerator : public RecordEnumerator, fleece::InstanceCounted {
     C4DocEnumerator(C4Database *database,
                     sequence_t since,
                     const C4EnumeratorOptions &options)
-    :_database(database),
-     _e(database->defaultKeyStore(), since, allDocOptions(options)),
-     _options(options)
+    :RecordEnumerator(database->defaultKeyStore(), since, recordOptions(options))
+    ,_database(database)
     { }
 
     C4DocEnumerator(C4Database *database,
                     const C4EnumeratorOptions &options)
-    :_database(database),
-     _e(database->defaultKeyStore(), allDocOptions(options)),
-     _options(options)
+    :RecordEnumerator(database->defaultKeyStore(), recordOptions(options))
+    ,_database(database)
     { }
 
-    void close() {
-        _e.close();
-    }
-
-    static RecordEnumerator::Options allDocOptions(const C4EnumeratorOptions &c4options) {
+    static RecordEnumerator::Options recordOptions(const C4EnumeratorOptions &c4options) {
         RecordEnumerator::Options options;
-        options.descending      = (c4options.flags & kC4Descending) != 0;
-        options.includeDeleted  = (c4options.flags & kC4IncludeDeleted) != 0;
+        if (c4options.flags & kC4Descending)
+            options.sortOption = kDescending;
+        else if (c4options.flags & kC4Unsorted)
+            options.sortOption = kUnsorted;
+        options.includeDeleted = (c4options.flags & kC4IncludeDeleted) != 0;
+        options.onlyConflicts  = (c4options.flags & kC4IncludeNonConflicted) == 0;
         if ((c4options.flags & kC4IncludeBodies) == 0)
             options.contentOption = kMetaOnly;
         return options;
     }
 
-    void setFilter(const EnumFilter &f)  {_filter = f;}
-
-    C4Database* database() const {return external(_database);}
-
-    bool next() {
-        do {
-            if (!_e.next())
-                return false;
-        } while (!useDoc());
-        return true;
-    }
-
     Retained<Document> getDoc() {
-        return _e ? _database->documentFactory().newDocumentInstance(_e.record()) : nullptr;
+        return *this ? _database->documentFactory().newDocumentInstance(record()) : nullptr;
```

Comment:
boolean test for `*this`?  Is there any way to be clearer about what this is actually checking?

Reply:
I know, it's kind of ugly. The class has an `operator bool` to test for EOF, but not a named method. I'll add a named method so it's clearer.